core/src/apps/management/authenticate_device.py

```diff
@@ -0,0 +1,48 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from trezor.messages import AuthenticateDevice, AuthenticityProof
+
+
+async def authenticate_device(msg: AuthenticateDevice) -> AuthenticityProof:
+    from trezor import utils
+    from trezor.crypto import optiga
+    from trezor.crypto.der import read_length
+    from trezor.crypto.hashlib import sha256
+    from trezor.messages import AuthenticityProof
+    from trezor.ui.layouts import confirm_action
+    from trezor.utils import BufferReader
+
+    from apps.common.writers import write_compact_size
+
+    await confirm_action(
+        "authenticate_device",
+        "Authenticate device",
+        description="Do you wish to verify the authenticity of your device?",
+    )
+
+    header = b"AuthenticateDevice:"
+    h = utils.HashWriter(sha256())
+    write_compact_size(h, len(header))
+    h.extend(header)
+    write_compact_size(h, len(msg.challenge))
+    h.extend(msg.challenge)
+
+    ctx = sha256()
+    ctx.update(msg.challenge)
```

Comment:
Am I blind or `ctx` is really not used anywhere?

Reply:
Fixed 9c99df71efbf53f23e1774a2cf3d017a230d5799. I changed from using `sha256()` directly to using `HashWriter` at some point and somehow failed to fully remove the  old code.